src/app/(with-main-header)/boards/[id]/Comment.tsx

```diff
@@ -14,13 +70,76 @@ export default function Comment({ data }: CommentProps) {
           className="w-11 lg:w-14 h-11 lg:h-14"
         />
       </Link>
-      <div className="flex flex-col lg:gap-1">
-        <Link href={`/users/${data.userId}`} className="w-fit">
-          <p className="font-semibold lg:text-xl">@{data.nickname}</p>
-        </Link>
-        <p className="text-site-darkgray-02 text-sm lg:text-base">
-          {data.context}
-        </p>
+
+      <div className="flex flex-col lg:gap-1 flex-grow">
+        <div className="flex justify-between items-center">
+          <Link href={`/users/${data.userId}`} className="w-fit">
+            <p className="font-semibold lg:text-xl">@{data.nickname}</p>
+            <p className="text-site-darkgray-02 text-sm lg:text-base">
+              {data.context}
+            </p>
+          </Link>
+
+          <div className="relative">
+            {(isCurrentUser ||
+              (isBoardAuthor &&
+                boardInfo?.boardType === "질문" &&
+                !isCurrentUser)) && (
+              <>
+                {isBoardAuthor &&
+                boardInfo?.boardType === "질문" &&
+                !isCurrentUser ? (
+                  isConfirmed ? (
+                    <Button
+                      className="bg-transparent px-0 lg:px-0 text-site-primary cursor-default"
+                      disabled
+                    >
+                      채택됨
+                    </Button>
+                  ) : (
+                    <Button
+                      onClick={() => setIsOpen((prev) => !prev)}
+                      className="bg-transparent px-0 lg:px-0"
+                    >
+                      <Icon MuiIcon={MoreHorizRoundedIcon} />
+                    </Button>
+                  )
+                ) : (
+                  <Button
+                    onClick={() => setIsOpen((prev) => !prev)}
+                    className="bg-transparent px-0 lg:px-0"
+                  >
+                    <Icon MuiIcon={MoreHorizRoundedIcon} />
+                  </Button>
+                )}
+              </>
+            )}
```

Comment:
```suggestion
            {isCurrentUser &&  moreIcon}
            {!isCurrentUser && isBoardAuthor &&  boardInfo?.boardType === "질문" && 채택버튼}
```

이런 식으로 구성하면 더 가독성이 좋을 것 같습니다! 그리고, 조건이 많은 경우에는 따로 변수로 빼는 것도 방법입니다!